cvat/apps/dataset_manager/formats/cvat.py

```diff
@@ -1282,39 +1282,39 @@ def load_anno(file_object, annotations):
                 shape['rotation'] = float(el.attrib.get('rotation', 0))
 
                 if el.tag == 'box':
-                    shape['points'].append(el.attrib['xtl'])
-                    shape['points'].append(el.attrib['ytl'])
-                    shape['points'].append(el.attrib['xbr'])
-                    shape['points'].append(el.attrib['ybr'])
+                    shape['points'].append(float(el.attrib['xtl']))
+                    shape['points'].append(float(el.attrib['ytl']))
+                    shape['points'].append(float(el.attrib['xbr']))
+                    shape['points'].append(float(el.attrib['ybr']))
                 elif el.tag == 'ellipse':
-                    shape['points'].append(el.attrib['cx'])
-                    shape['points'].append(el.attrib['cy'])
-                    shape['points'].append("{:.2f}".format(float(el.attrib['cx']) + float(el.attrib['rx'])))
-                    shape['points'].append("{:.2f}".format(float(el.attrib['cy']) - float(el.attrib['ry'])))
+                    shape['points'].append(float(el.attrib['cx']))
+                    shape['points'].append(float(el.attrib['cy']))
+                    shape['points'].append(float("{:.2f}".format(float(el.attrib['cx']) + float(el.attrib['rx']))))
+                    shape['points'].append(float("{:.2f}".format(float(el.attrib['cy']) - float(el.attrib['ry']))))
                 elif el.tag == 'mask':
-                    shape['points'] = el.attrib['rle'].split(',')
-                    shape['points'].append(el.attrib['left'])
-                    shape['points'].append(el.attrib['top'])
-                    shape['points'].append("{}".format(int(el.attrib['left']) + int(el.attrib['width']) - 1))
-                    shape['points'].append("{}".format(int(el.attrib['top']) + int(el.attrib['height']) - 1))
+                    shape['points'] = list(map(float, el.attrib['rle'].split(',')))
+                    shape['points'].append(float(el.attrib['left']))
+                    shape['points'].append(float(el.attrib['top']))
+                    shape['points'].append(float("{}".format(int(el.attrib['left']) + int(el.attrib['width']) - 1)))
+                    shape['points'].append(float("{}".format(int(el.attrib['top']) + int(el.attrib['height']) - 1)))
```

Comment:
By the way, for a mask, all these numbers are integers. They just can't be floats.

Reply:
Good catch!